### PR TITLE
Refactoring before tests for Source

### PR DESCRIPTION
**Pull request by lukaszwawrzyk**

### CODE REVIEW

File: ci/update-checker/src/main/scala/com/twitter/intellij/updatechecker/UpdateChecker.scala [hunk 1]
```diff
@@ -10,8 +10,8 @@ import org.virtuslab.ideprobe.dependencies.IntelliJVersion
 import org.virtuslab.ideprobe.dependencies.Plugin.Versioned
 import pureconfig.generic.auto._
 
-import scala.jdk.CollectionConverters._
-import scala.math.Ordering.Implicits.seqOrdering
+import scala.collection.JavaConverters._
```
Comment (odisseus) [inline ci/update-checker/src/main/scala/com/twitter/intellij/updatechecker/UpdateChecker.scala:13]: These are deprecated in Scala 2.13.

File: tests/src/test/resources/pants.conf [hunk 2]
```diff
@@ -3,12 +3,27 @@ include required("versions.conf")
...
     launch.timeout = "2 minutes"
-    headless = false # make all pants test run in ui mode by default for more accurate results
-    check.freezes = true # fail test in case of detecting freeze
+    vmOptions = [
+      "-Dvcs.log.index.git=false",
+      "-Dgit.process.ignored=false",
+      "-Ddotty.highlighting.compiler.errors.in.editor=false",
+      "-Dvcs.root.detector.folder.depth=1",
+      "-Xms2g",
+      "-Xmx12g",
```
Comment (odisseus) [inline tests/src/test/resources/pants.conf:12]: The standard runner of Github Actions doesn't [have](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources) that much RAM.

File: tests/src/test/resources/versions.conf [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
+  intellij {
+    build = "203.5784.10"
+  }
   plugins {
-    pythonCommunity = { id = "PythonCore", version = "203.4203.26" }
-    scala = { id = "org.intellij.scala", version = "2020.3.397", channel = "nightly" }
-    pants = {
+    pythonCommunity = { id = "PythonCore", version = ${versions.intellij.build} }
+    pythonUltimate  = { id = "Pythonid", version = ${versions.intellij.build} }
+    scala           = { id = "org.intellij.scala", version = "2020.3.442", channel = "nightly" }
```
Comment (odisseus) [inline tests/src/test/resources/versions.conf:8]: There is already a version [2020.3.628](https://plugins.jetbrains.com/plugin/1347-scala/versions/nightly).

File: ci/update-checker/src/main/scala/com/twitter/intellij/updatechecker/UpdateChecker.scala (hunk 1 above)
Comment (lukaszwawrzyk) [reply ci/update-checker/src/main/scala/com/twitter/intellij/updatechecker/UpdateChecker.scala:13]: we need code to be 2.12 and 2.13 compatible. These are 2 lines, won't hurt us and are the. easiest way to make it work.

File: tests/src/test/resources/pants.conf (hunk 2 above)
Comment (lukaszwawrzyk) [reply tests/src/test/resources/pants.conf]: I just copied settings that we use. So let's roll that back.

File: tests/src/test/resources/versions.conf (hunk 3 above)
Comment (lukaszwawrzyk) [reply tests/src/test/resources/versions.conf:8]: I tested with some newer version, it was failing. The purpose of this PR is mostly to merge all other changes. Testing versions can be done with automatic bump tool.

**Pull request merged**